app/src/main/java/org/mozilla/fenix/trackingprotection/TrackingProtectionPanelView.kt

```diff
@@ -0,0 +1,217 @@
+            View.VISIBLE else View.GONE
+        tracking_content.visibility = if (bucketedTrackers[TRACKING_CONTENT]?.isNotEmpty() == true)
+            View.VISIBLE else View.GONE
+        cryptominers.visibility = if (bucketedTrackers[CRYPTOMINERS]?.isNotEmpty() == true)
+            View.VISIBLE else View.GONE
+    }
+
+    private fun setCategoryClickListeners() {
+        social_media_trackers.setOnClickListener {
+            interactor.openDetails(SOCIAL_MEDIA_TRACKERS)
+        }
+        fingerprinters.setOnClickListener {
+            interactor.openDetails(FINGERPRINTERS)
+        }
+        cross_site_tracking.setOnClickListener {
+            interactor.openDetails(CROSS_SITE_TRACKING_COOKIES)
+        }
+        tracking_content.setOnClickListener {
+            interactor.openDetails(TRACKING_CONTENT)
+        }
+        cryptominers.setOnClickListener {
+            interactor.openDetails(CRYPTOMINERS)
+        }
+    }
+
+    private fun setUIForDetailsMode(
+        category: TrackingProtectionCategory
+    ) {
+        normal_mode.visibility = View.GONE
+        details_mode.visibility = View.VISIBLE
+        category_title.text = context.getString(category.title)
+        val stringList = bucketedTrackers[category]?.joinToString("\n")
+        blocking_text_list.text = stringList
+        category_description.text = context.getString(category.description)
+        details_back.setOnClickListener {
+            interactor.onBackPressed()
+        }
+    }
+
+    private fun getHashMapOfTrackersForCategory(
+        list: List<Tracker>
+    ): HashMap<TrackingProtectionCategory, List<String>> {
+        val hashMap = HashMap<TrackingProtectionCategory, List<String>>()
+        for (item in list) {
+            when {
+                item.categories.contains(Tracker.Category.Cryptomining) -> hashMap[CRYPTOMINERS] =
+                    (hashMap[CRYPTOMINERS]
+                        ?: listOf()).plus(item.url.getHostFromUrl() ?: item.url)
+                item.categories.contains(Tracker.Category.Fingerprinting) -> hashMap[FINGERPRINTERS] =
+                    (hashMap[FINGERPRINTERS]
+                        ?: listOf()).plus(item.url.getHostFromUrl() ?: item.url)
+                item.categories.contains(Tracker.Category.Social) -> hashMap[SOCIAL_MEDIA_TRACKERS] =
+                    (hashMap[SOCIAL_MEDIA_TRACKERS] ?: listOf()).plus(
+                        item.url.getHostFromUrl() ?: item.url
+                    )
+                item.categories.contains(Tracker.Category.Content) -> hashMap[TRACKING_CONTENT] =
+                    (hashMap[TRACKING_CONTENT]
+                        ?: listOf()).plus(item.url.getHostFromUrl() ?: item.url)
+                else ->
+                    // TODO this is not right but looking at the doc I'm not sure what they want.
+                    //  Ask Arturo where Ad/Analytic are supposed to go?
```

Comment:
@Amejia481 could you help me understand that document as it relates to this API with the categories? :) I understood until the last step or so

Reply:
For the last one we are waiting for GV to [proving us an API that tells which categories were blocked as cookie-based. ](https://bugzilla.mozilla.org/show_bug.cgi?id=1567268)